spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -391,6 +391,44 @@ def count
       expect(counter.count).to eq(2)
     end
 
+    it "is threadsafe" do
+      value_queue = Queue.new
+      values      = []
+      RSpec.describe do
+        # a let block that will block
+        let(:memoize_threadsafe) { value_queue.shift }
+
+        example do
+          thread_blocked = Queue.new
+
+          # blocked, waiting on let to return
+          thread1 = Thread.new do
+            thread_blocked << true
+            values << memoize_threadsafe
+          end
+          thread_blocked.shift
+
+          # blocked, waiting on let to return
+          thread2 = Thread.new do
+            thread_blocked << true
+            values << memoize_threadsafe
+          end
+          thread_blocked.shift
+
+          # provide the values they are blocking for
+          value_queue << :value1
+          value_queue << :value2
+
+          # wait for them to finish
+          thread1.join
+          thread2.join
+        end
+      end.run
+
+      expect(values.size).to eq 2
+      expect(values[0]).to eq values[1]
```

Comment:
As of right now, no, b/c w/ current implementation, it could also be `[:value2, :value2]` I could maybe further synchronize it so that you know when the threads run in relation to each other, and then get this kind of confidence, but it could get pretty verbose without creating abstractions. Which I don't really mind doing, I had thought about it, even, but it just seemed heavy since it's just for this one test.